packages/itwinui-css/src/input/input.scss

```diff
@@ -108,7 +108,11 @@
 
   > :first-child {
     grid-area: 1 / -1;
-    padding-inline-end: var(--iui-component-height); // Width of the end-icon button
+
+    // When there is an end-icon button (i.e. first-child is not the only child)
+    &:not(:only-child) {
+      padding-inline-end: var(--iui-component-height); // Width of the end-icon button
+    }
```

Comment:
To avoid adding padding when no end-icon is present in `.iui-input-with-icon`. This fixes the [Table → Filter story](https://itwin.github.io/iTwinUI/react/?story=table--filters&theme=dark) test.

|Without this change|With this change|
|-|-|
|![Screenshot 2024-01-11 at 3 41 08 PM](https://github.com/iTwin/iTwinUI/assets/45748283/719688e2-f428-4d1b-8698-eade9363d8f8)|![Screenshot 2024-01-11 at 3 38 57 PM](https://github.com/iTwin/iTwinUI/assets/45748283/96858d77-4057-4dde-91ca-0d714b8d7a6a)|

Reply:
we can make this more robust using `:has`.

```scss
&:has(+ .iui-end-icon) {
  padding-inline-end: var(--iui-component-height);
}

@supports not selector(:has(+ *)) {
  &:not(:only-child) {
    padding-inline-end: var(--iui-component-height);
  }
}
```

it is not good to rely on selectors `:first-child`/`:only-child` etc

Reply:
Ohh, I see, changed (https://github.com/iTwin/iTwinUI/pull/1753/commits/b7f876304305f55a805493cc305d8ac16a9d97b5)